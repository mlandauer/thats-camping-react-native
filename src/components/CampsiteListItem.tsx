import * as React from 'react'
import {
  StyleSheet,
  Text,
  View,
  TouchableHighlight
} from 'react-native'

import shortenName from '../libs/shortenName'
import * as TextFormatter from '../libs/TextFormatter'
import Star from '../components/Star'

interface Props {
  campsiteName: string;
  parkName: string;
  starred: boolean;
  distance: number | undefined;
  bearing: number | undefined;
  onPress: () => void;
}

export default function CampsiteListItem(props: Props) {
  return (
    <TouchableHighlight onPress={props.onPress}>
      <View style={styles.container}>
        <View style={{ flex: 1 }}>
          <Text style={styles.campsiteName}>{shortenName(props.campsiteName)}</Text>
          <Text numberOfLines={1} style={styles.parkName}>{props.parkName}</Text>
        </View>
        <View style={{ flexDirection: 'row' }}>
          <Text style={styles.distanceAndBearing}>{TextFormatter.distanceText(props.distance)} {TextFormatter.bearingText(props.bearing)}</Text>
          <HidingStar starred={props.starred} size={24} />
        </View>
      </View>
    </TouchableHighlight>
  )
}

function HidingStar(props: { starred: boolean, size: number }) {
  if (props.starred) {
    return (
      <Star starred={props.starred} size={props.size} />
    )
  } else {
    return null
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 16,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: 'white'
  },
  campsiteName: {
    fontSize: 20,
    fontWeight: "500" as "500"
  },
  parkName: {
    fontSize: 20,
    color: '#aaa'
  },
  distanceAndBearing: {
    fontSize: 20,
    color: '#aaa'
  }
})
