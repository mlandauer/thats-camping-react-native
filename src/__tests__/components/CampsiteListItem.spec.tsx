import * as renderer from 'react-test-renderer'
import * as React from 'react'

import CampsiteListItem from '../../components/CampsiteListItem'

it('renders correctly', () => {
  const tree = renderer.create(
    <CampsiteListItem campsiteName="Lovely campsite" parkName="Peaceful area" starred={false} distance={1000} bearing={180}/>
  ).toJSON();
  expect(tree).toMatchSnapshot()
})
