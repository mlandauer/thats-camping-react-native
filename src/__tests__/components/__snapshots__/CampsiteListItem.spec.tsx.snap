// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
          }
        }
      >
        Lovely campsite
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#aaa",
            "fontSize": 20,
          }
        }
      >
        Peaceful area
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#aaa",
            "fontSize": 20,
          }
        }
      >
        1000 m
         
        S
      </Text>
    </View>
  </View>
</View>
`;
